torchgen/model.py

```diff
@@ -1293,7 +1304,7 @@ def kind(self) -> SchemaKind:
         if is_inplace:
             return SchemaKind.inplace
         elif is_out:
-            return SchemaKind.out
+            return SchemaKind.scratch if is_scratch else SchemaKind.out
```

Comment:
yep, I was thinking a flattened if statement impose some limitation that `elif is_scratch:` must comes before `elif isout:` .  But the assertion you added inspired me. I can add:
```
assert !is_scratch, "..."
```
in the block for `elif is_out:` to avoid misuse.